tests/Unit/Evolution/Systems/Cce/AnalyticSolutions/Test_TeukolskyWave.cpp

```diff
@@ -0,0 +1,297 @@
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/Evolution/Systems/Cce/AnalyticSolutions/AnalyticDataHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce::Solutions {
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.Cce.TeukolskyWave", "[Unit][Cce]") {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<double> radius_dist{5.0, 10.0};
+  UniformCustomDistribution<double> parameter_dist{0.5, 2.0};
+  const double extraction_radius = radius_dist(gen);
+  const size_t l_max = 16;
+  // use a low frequency so that the dimensionless parameter in the metric is ~1
+  const double amplitude = parameter_dist(gen) / 10.0;
+  const double duration = 1.0;
+  const TeukolskyWave boundary_solution{extraction_radius, amplitude, duration};
+  const double time = duration + extraction_radius;
+
+  const auto boundary_data = boundary_solution.variables(
+      l_max, time,
+      tmpl::list<
+          Tags::Dr<Tags::CauchyCartesianCoords>,
+          gr::Tags::SpacetimeMetric<3, ::Frame::Inertial, DataVector>,
+          ::Tags::dt<
+              gr::Tags::SpacetimeMetric<3, ::Frame::Inertial, DataVector>>,
+          GeneralizedHarmonic::Tags::Phi<3, ::Frame::Inertial>, Tags::News>{});
+  const auto& spacetime_metric =
+      get<gr::Tags::SpacetimeMetric<3, ::Frame::Inertial, DataVector>>(
+          boundary_data);
+
+  SphericaliCartesianJ jacobian{get<0, 0>(spacetime_metric).size()};
+  boundary_solution.jacobian(make_not_null(&jacobian), l_max);
+
+  const auto spherical_metric =
+      TestHelpers::cartesian_metric_to_spherical_metric(spacetime_metric,
+                                                        jacobian);
+
+  // so that the comparisons with zero succeed
+  Approx scale_approx =
+      Approx::custom()
+          .epsilon(std::numeric_limits<double>::epsilon() * 1.0e4)
+          .scale(1.0);
+
+  DataVector sin_theta{get<0, 0>(spherical_metric).size()};
+  DataVector cos_theta{get<0, 0>(spherical_metric).size()};
+  const auto& collocation = Spectral::Swsh::cached_collocation_metadata<
+      Spectral::Swsh::ComplexRepresentation::Interleaved>(l_max);
+  for (const auto& collocation_point : collocation) {
+    sin_theta[collocation_point.offset] = sin(collocation_point.theta);
+    cos_theta[collocation_point.offset] = cos(collocation_point.theta);
+  }
+  // this should give  B * (-3 sin theta * cos theta)
+  const DataVector expected_b_coefficient_dependence =
+      6.0 * amplitude *
+      (3.0 + square(extraction_radius) *
+                 (-6.0 + extraction_radius * (3.0 + 2.0 * extraction_radius))) /
+      (M_E * pow<6>(extraction_radius)) * sin_theta * cos_theta;
```

Comment:
also setting k=u is bad for testing;  this seems ideal for testing with python...